kymatio/scattering1d/frontend/numpy_frontend.py

```diff
@@ -22,20 +22,6 @@ def scattering(self, x):
                 'Input tensor x should have at least one axis, got {}'.format(
                     len(x.shape)))
 
-        if not self.out_type in ('array', 'list'):
-            raise RuntimeError("The out_type must be one of 'array' or 'list'.")
-
-        if not self.average and self.out_type == 'array' and self.vectorize:
-            raise ValueError("Options average=False, out_type='array' and "
-                             "vectorize=True are mutually incompatible. "
-                             "Please set out_type to 'list' or vectorize to "
-                             "False.")
```

Comment:
I think we wanna keep this non?

Reply:
these errors now live in `core/scattering1d.py`, to avoid repeating the text across frontends:

```python
    if out_type not in ['array', 'dict', 'list']:
        raise RuntimeError("'out_type' must either be 'array', 'dict', or 'list'.")

    if out_type=="array" and not average:
        raise RuntimeError("out_type=='array' and average are mutually incompatible.")
```

Reply:
we should move these lines to the base frontend object then

Reply:
but which method of `ScatteringBase1D`? it does not have a `scattering` method. Should i introduce a `check_out_type` method or something like that? Or call the method `scattering` and call `super().scattering(self, input) ?

Reply:
Yeah, introduce `check_out_type` for now. what do you think @janden ?

Reply:
i like this `check_out_type` idea

Reply:
implemented in ff8344a